src/ng/http.js

```diff
@@ -1130,28 +1130,61 @@ function $HttpProvider() {
 
 
     function buildUrl(url, params) {
-      if (!params) return url;
+      params = $httpUrlParams.serialize(params);
+      if (params) {
+        url += ((url.indexOf('?') == -1) ? '?' : '&') + params;
+      }
+      return url;
+    }
+  }];
+}
+
+/**
+ * @ngdoc provider
+ * @name $httpUrlParamsProvider
+ * @description
+ * Use `$httpUrlParamsProvider` to change the default behavior of the {@link ng.$httpUrlParams $httpUrlParams} service.
+ * */
+function $HttpUrlParamsProvider() {
```

Comment:
Maybe it is better to move this service (and maybe also the relevant unit tests) to a different file?